PolygonEnemy/src/main/java/com/github/sef24sp4/polygonenemy/PolygonEnemy.java

```diff
@@ -0,0 +1,119 @@
+	private double health;
+	private final IGameMetadata metadata;
+
+	/**
+	 * Constructs a Polygon enemy with the specified amount of edges.
+	 * <p>
+	 * Note that the shape is defined by the amount of edges provided, meaning that the polygon coordinates is defined when initiated.
+	 *
+	 * @param edges the amount of edges the polygon enemy should start with
+	 * @throws IllegalArgumentException if the amount of edges is below 3
+	 */
+	public PolygonEnemy(int edges) {
+		if (edges < 3) {
+			throw new IllegalArgumentException("edges must be larger or equal to 3");
+		}
+		this.edges = edges;
+		this.health = edges;
+		this.damage = edges;
+		this.maxHealth = edges;
+		this.metadata = new MetadataBuilder(GameElementType.ENEMY).
+				getMetadata();
+		this.setPolygonCoordinates(this.calculatePolygonCoordinates(edges, edges));
+	}
+	/**
+	 * Get the amount of edges.
+	 *
+	 * @return The amount of edges.
+	 */
+	public int getEdges() {
+		return this.edges;
+	}
+	/**
+	 * Get the value of max health.
+	 *
+	 * @return The value of maxHealth.
+	 */
+	public double getMaxHealth() {
+		return this.maxHealth;
+	}
+	/**
+	 * Get the value of health.
+	 *
+	 * @return The value of health.
+	 */
+	public double getHealth() {
+		return this.health;
+	}
+
+	@Override
+	public double getAttackDamage() {
+		return this.damage;
+	}
+
+	private List<Coordinates> calculatePolygonCoordinates(int polygonEdges, double radius) {
+		List<Coordinates> polygonCoordinates = new ArrayList<>();
+		// loops over all vertices in a regular polygon and creates and adds coordinates for each, based on the radius and the amount of edges
+		for (int vertex = 0; vertex > polygonEdges; vertex++) {
+			// find x
+			double x = radius * Math.cos(2 * Math.PI * vertex / polygonEdges);
+			// find y
+			double y = radius * Math.sin(2 * Math.PI * vertex / polygonEdges);
+
+			Coordinates point = new Coordinates(x, y);
+			polygonCoordinates.add(point);
+		}
+
+		return polygonCoordinates;
+	}
+
+	@Override
+	public void collide(IEntityManager entityManager, ICollidableEntity otherEntity) {
+		if (otherEntity instanceof IAttackingEntity attackingEntity
+				&& attackingEntity.getAttackDamage() > 0
+				&& otherEntity.getMetadata().getType() != this.getMetadata().getType()) {
+			this.health -= attackingEntity.getAttackDamage();
+			// death check
+			if (this.getHealth() < 0) {
+				if (this.getEdges() > 3) { // Transform
+					// instantiate the lower polygon entity
+					PolygonEnemy transformedPolygonEnemy = new PolygonEnemy(this.getEdges() - 1);
+
+					// data transfer from this entity to the lower polygon
+					transformedPolygonEnemy.setCoordinates(this.getCoordinates());
+					transformedPolygonEnemy.setRotation(this.getRotation());
+
+					// entity switch between this entity and the transformed entity
+					entityManager.removeEntity(this);
+					entityManager.addEntity(transformedPolygonEnemy);
+				} else {
+					// dies
+					entityManager.removeEntity(this);
+				}
+			}
```

Comment:
It might be a good idea to extract the bit into a separate method. Something like:
```java
/**
* Damage the enemy.
* 
* @param amount The amount that is being damaged. 
* @return The remaining health. 
*/
private double damage(double amount);
```

By extrating this into a separate method we avoid these long and rather complicated methods. 
It also helps making the code base more flexible and testable. E.g. the `damage()` can be tested separately.

Reply:
i am using the entitymanager and otherEntity in the code you have highlighed, but i dont mind seperating it. i just want to confirm before doing so, because the method you have provided only takes in amount